test/e2e/app.test.js

```diff
@@ -0,0 +1,57 @@
+const assert = require('assert')
+const electron = require('electron')
+const path = require('path')
+const webdriverio = require('webdriverio')
+
+let app, started, crashed
+
+const sleep = time => new Promise(resolve => setTimeout(resolve, time))
+
+function launch(t) {
+  if (!started) {
+    started = new Promise(async resolve => {
+      const options = {
+        host: 'localhost',
+        port: 9515,
+        desiredCapabilities: {
+          browserName: 'chrome',
+          chromeOptions: {
+            binary: electron,
+            args: ['app='+path.join(__dirname, '..', '..', 'app', 'main.js')]
+          }
+        }
+      }
+      app = webdriverio.remote(options)
+
+      let initialElement = ".page-body"
+      await startApp(app, initialElement)
+      resolve(app)
+
+    })
+  }
+
+  return started
+}
+
+async function startApp(app, awaitingSelector) {
+    await app.init()
+    const isStarted = await app.getHTML(awaitingSelector)
+    assert.ok(isStarted)
+}
+
+describe('Application launch', async () => {
+  beforeAll(async () => {
+    await launch();
+  })
+
+  // afterAll(() => app.end())
+
+  it('Sync Decrediton', async () => {
+    await sleep(5000);
+    await app.click('//*[@id="root"]/div[1]/div/div/div/div/div[5]/div/div/div[1]/div[5]/span')
```

Comment:
This is one of the main challenges I had when working with selenium, though it was somewhat less prominent pre-react because we used classes and IDs more extensively: how do you locate an element to perform an action or check a state.

Don't know what the best practices are with react (post some links if you find anything about it). But this style will _definitely_ be very brittle: a single move of a component in the hierarchy is going to break this (and possibly many other) tests.

Maybe we should start adding a `testID` attribute to components which will be used in a test, such that this id is generated when building a development version of decrediton but not on production. But this will increase the verbosity of the code.

Reply:
Right now I am using xpath, but yeah, you are right, this can be a problem when testing.

Here http://codefuse.io/ui-test-object-identification-with-id-css-xpath/ there is a text explain the pro's and con's in using ids, css selectors and xpath.

I believe the proper order would be: Ids, then css selectors, and then in last case xpaths, when can not use both, because this is the more suggestive to changes. But beyond that I believe there is nothing we can do beyond fixing the test when doing changes on decrediton. 

I do not think would be good changing all the code to add testID all over the code, just for that. Maybe adding ids?